src/NosCore.GameObject/Providers/InventoryService/InventoryService.cs

```diff
@@ -474,6 +474,40 @@ public InventoryItemInstance RemoveItemAmountFromInventory(short amount, Guid id
             return null;
         }
 
+        public List<InventoryItemInstance> RemoveItemAmountFromInventoryByVNum(short amount, short itemVNum)
+        {
+            List<InventoryItemInstance> result = new List<InventoryItemInstance>();
+            var stayAmount = amount;
+            foreach (var itemInstance in this.Select(s => s.Value)
```

Comment:
Even though your compiler optimizes it, you should store the LINQ query in a variable in order to keep clear code, atm reading the condition of the loop is quite harsh